query/src/main/java/org/infinispan/query/impl/massindex/DistributedExecutorMassIndexer.java

```diff
@@ -51,7 +52,12 @@
    };
 
    public DistributedExecutorMassIndexer(AdvancedCache<?, ?> cache) {
+      this(cache, false);
+   }
+
+   public <K, V> DistributedExecutorMassIndexer(AdvancedCache<K, V> cache, boolean localOnly) {
       this.cache = cache;
+      this.localOnly = localOnly;
```

Comment:
since we are keeping the AdvanceCache changes, I would remove the field and use the method

Reply:
ok

Reply:
actually not OK. Unfortunately the Indexer is a component from the registry and the flagged cache is an instance, so I need a decorator with local field

Reply:
the instance in the registry is fine. you're creating a new instance when the cache is flagged in `getIndexer()` that is why I think the boolean is not required:
```
      if (cache.getAdvancedCache().hasFlag(Flag.CACHE_MODE_LOCAL)) {
         return new DistributedExecutorMassIndexer(advancedCache, true);
      }
```
ps. just noticed you can replace  `cache.getAdvancedCache()` with `advancedCache` :)